Conditionally install locking callbacks for OpenSSL.

We are a library. The application must have already done so in
its main thread (and, if it hasn't, it should be fixed to do so)
so test to see if a callback exists and do nothing if so.

diff --git a/libraries/libldap/tls_o.c b/libraries/libldap/tls_o.c
index b0277df..092f20b 100644
--- a/libraries/libldap/tls_o.c
+++ b/libraries/libldap/tls_o.c
@@ -106,11 +106,13 @@ static void tlso_thr_init( void )
 {
 	int i;
 
-	for( i=0; i< CRYPTO_NUM_LOCKS ; i++ ) {
-		ldap_pvt_thread_mutex_init( &tlso_mutexes[i] );
+	if ( !CRYPTO_get_locking_callback() ) {
+		for( i=0; i< CRYPTO_NUM_LOCKS ; i++ ) {
+			ldap_pvt_thread_mutex_init( &tlso_mutexes[i] );
+		}
+		CRYPTO_set_locking_callback( tlso_locking_cb );
+		CRYPTO_set_id_callback( tlso_thread_self );
 	}
-	CRYPTO_set_locking_callback( tlso_locking_cb );
-	CRYPTO_set_id_callback( tlso_thread_self );
 }
 #endif /* LDAP_R_COMPILE */
 #else
